### PR TITLE
Towerinfov4 slimmed down tower objects for high lumi running 

### PR DESCRIPTION
**Pull request by bseidlit**
[comment]: <> (Please tell us something about this pull request)

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

### CODE REVIEW

File: offline/packages/CaloBase/TowerInfov4.h
Comment (pinkenburg) [inline offline/packages/CaloBase/TowerInfov4.h:1]: Please remove TMath, no need to use it here (log is part of c++ cmath) and follow our include file conventions:
https://wiki.sphenix.bnl.gov/index.php?title=Codingconventions#Include_Files

**Comment by pinkenburg** [comment]
The clang error is from another PR, if you merge this with the current master branch it'll be fixed

**Comment by bseidlit** [comment]
@pinkenburg  I switched to std log, it compiled and it ran a test.  Do you think we can just merge this so others can test the new objects?

**Comment by pinkenburg** [comment]
We probably need one more change - the memory layout of the CaloTowerv4 is inefficient. It inherits from CaloTowerv1 which uses
 private:
  short _time = 0;
  float _energy = 0;
which should have never been committed this way (it leaves a 16 bit hole in the middle of the data). Now CaloTowerv4 adds to this and it's layout is
 private:
  short _time = 0;
  float _energy = 0;
 uint8_t _chi2 = 0;
  uint8_t _status = 0;
adding another 16 bit hole at the end. We should drop the inheritance from v1 and use

private:
 float _energy = 0;
short _time = 0;
uint8_t _chi2 = 0;
 uint8_t _status = 0;

which now lines this nicely up along 32 bit boundaries (or event 64 bit)
and maybe drop the _'s from the variable names which are just painful when T->Drawing() interactively

**Comment by bseidlit** [comment]
@pinkenburg  I made the switches.  The resulting DST was the exact same size, not sure if that was expected but OK.

**Comment by pinkenburg** [comment]
I didn't expect the size to change but for the cpu I was thinking that it's easier for memory access by the cpu if data is aligned (they typically fetch 32 or 64 bit words in one shot independent of the "real" size). I don't know if this has a real effect but it's free - so why not try? In terms of storage - root stores every variable separately in a "basket" which is then compressed, for that the memory layout doesn't matter.

**Comment by bseidlit** [comment]
OK, yeah Ii was wondering if it was something like that.  Yeah no sweat, certainly free and an easy change. I also never understood the leading underscore (no ones got time for that!!)

**Pull request merged**